pkg/kubelet/kuberuntime/kuberuntime_container_test.go

```diff
@@ -493,6 +501,26 @@ func TestLifeCycleHook(t *testing.T) {
 			t.Errorf("CMD PostStart hook was not invoked")
 		}
 	})
+
+	// Execute a long-term PreStop CMD and complete it within the expected time
+	t.Run("Long-Term-PreStop-CMDExec", func(t *testing.T) {
+		ctx := context.Background()
+
+		//one second of gracePeriod is reserved to trigger PreStop
+		gracePeriod := int64(1)
+
+		startTime := time.Now()
+		lcHanlder = lifecycletest.NewFakeHandlerRunner()
+		m.runner = lcHanlder
+
+		testPod.Spec.Containers[0].Lifecycle = cmdLongTermLifeCycle
+		m.killContainer(ctx, testPod, cID, "foo", "testKill", "", &gracePeriod)
+
+		if time.Since(startTime) > time.Duration(gracePeriod+minimumGracePeriodInSeconds)*time.Second {
+			t.Error("When gracePeriod is 0, PreStop exec should complete in 3s")
+		}
+	})
```

Comment:
I think this cannot simulate https://github.com/kubernetes/kubernetes/issues/115817 as `runtime.StopContainer` always returns `contaienr not found error` immediately. (This test succeeds without the fix as well.)

I wrote [an e2e test](https://github.com/kubernetes/kubernetes/pull/116911) to simulate this scenario as it may be too complex to write it in a unit test. Could you cherry-pick that commit or write your own and remove this unit test if you don't mind?

Reply:
> I think this cannot simulate #115817 as `runtime.StopContainer` always returns `contaienr not found error` immediately. (This test succeeds without the fix as well.)
> 
> I wrote [an e2e test](https://github.com/kubernetes/kubernetes/pull/116911) to simulate this scenario as it may be too complex to write it in a unit test. Could you cherry-pick that commit or write your own and remove this unit test if you don't mind?

Okay, I will delete this unit test and cherry-pick your commit

Reply:
you just cherry-picked outdated commit (sorry, I force-pushed it)

Reply:
> you just cherry-picked outdated commit (sorry, I force-pushed it)

Already cherry-pick the latest commit

Reply:
> you should cherry-pick the test again as I fixed a lint error.

Okay